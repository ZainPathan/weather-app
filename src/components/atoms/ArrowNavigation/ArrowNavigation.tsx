import React from 'react';
import {
    ArrowBack as ArrowBackIcon,
    ArrowForward as ArrowForwardIcon
} from '@material-ui/icons';
import { Grid } from '@material-ui/core';
import useArrowNavigationStyles from './styles/ArrowNavigation.style';
// import clsx from 'clsx';

interface ArrowNavigationProps {
    handleNavigation: Function,
    currentCardIndex: number,
    maxCardCount: number
}


const ArrowNavigation = (props: ArrowNavigationProps) => {

    const classes = useArrowNavigationStyles(props);

    const handleArrowClick = (event: React.MouseEvent, direction: string) => {
        // console.log('handleArrowClick event: ', event);
        // console.log('handleArrowClick direction: ', direction);
        switch(direction) {
            case 'back':
                if((props.currentCardIndex === 0)) {
                    event.preventDefault();
                    return null;
                }
                break;
            case 'forward':
                if((props.currentCardIndex === (props.maxCardCount - 1))) {
                    event.preventDefault();
                    return null;
                }
                break;
        }
        props.handleNavigation(direction);
    }

    // console.log('props.currentCardIndex: ', props.currentCardIndex);
    // console.log('props.currentCardIndex !== 0: ', props.currentCardIndex !== 0);
    // console.log('props.maxCardCount: ', props.maxCardCount);
    // console.log('props.currentCardIndex === (props.maxCardCount - 1): ',props.currentCardIndex === (props.maxCardCount - 1));

    return (
        <Grid
            item
            xs={12}
            className={classes.arrowNavigation}
        >
            {
                <ArrowBackIcon
                    data-testid="back-arrow"
                    className={
                        (props.currentCardIndex === 0) ?
                            classes.hideIcon : ''
                    }
                    onClick={(e: React.MouseEvent) => handleArrowClick(e, 'back')}
                />
            }
            {
                <ArrowForwardIcon
                    data-testid="forward-arrow"
                    className={
                        (props.currentCardIndex === (props.maxCardCount - 1)) ?
                            (classes.hideIcon) : ''
                    }
                    onClick={(e: React.MouseEvent) => handleArrowClick(e, 'forward')}
                />
            }
        </Grid>
    )
};

export default ArrowNavigation;