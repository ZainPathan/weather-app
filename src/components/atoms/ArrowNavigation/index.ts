import ArrowNavigation from './ArrowNavigation';

export {
    ArrowNavigation
};

export default ArrowNavigation;