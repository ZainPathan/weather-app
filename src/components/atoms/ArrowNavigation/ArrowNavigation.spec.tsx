import React from 'react';
import renderer from 'react-test-renderer';
import { fireEvent, render } from '@testing-library/react';
import ArrowNavigation from "./ArrowNavigation";

describe('ArrowNavigation component tests', () => {

    const mockHandleNavigation = jest.fn();
    const mockCurrentCardIndex = 1;
    const mockMaxCardCount = 6;

    it('renders properly', () => {
        const tree = renderer
            .create(
                <ArrowNavigation
                    handleNavigation={mockHandleNavigation}
                    currentCardIndex={mockCurrentCardIndex}
                    maxCardCount={mockMaxCardCount}
                />
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders without Back Arrow when on first card', () => {
        const mockCurrentCardIndexFirst = 0;
        const { container, getByTestId, debug } = render(
            <ArrowNavigation
                handleNavigation={mockHandleNavigation}
                currentCardIndex={mockCurrentCardIndexFirst}
                maxCardCount={mockMaxCardCount}
            />
        );
        // debug();
        // expect(container.querySelector('.MuiSvgIcon-root')).toHaveClass('makeStyles-hideIcon-4');
        expect(getByTestId('back-arrow')).toHaveClass('makeStyles-hideIcon-4');
    });

    it('renders without Forward Arrow when on last card', () => {
        const mockCurrentCardIndexLast = mockMaxCardCount - 1;
        const { container, getByTestId } = render(
            <ArrowNavigation
                handleNavigation={mockHandleNavigation}
                currentCardIndex={mockCurrentCardIndexLast}
                maxCardCount={mockMaxCardCount}
            />
        );
        expect(getByTestId('forward-arrow')).toHaveClass('makeStyles-hideIcon-4');
    });

    it('handleNavigation() is called', () => {
        const { container, getByTestId, debug } = render(
            <ArrowNavigation
                handleNavigation={mockHandleNavigation}
                currentCardIndex={mockCurrentCardIndex}
                maxCardCount={mockMaxCardCount}
            />
        );
        fireEvent.click(getByTestId('back-arrow'));
        expect(mockHandleNavigation).toHaveBeenCalled();
    });
});