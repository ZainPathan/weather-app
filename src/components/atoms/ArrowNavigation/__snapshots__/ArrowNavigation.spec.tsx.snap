// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArrowNavigation component tests renders properly 1`] = `
<div
  className="MuiGrid-root makeStyles-arrowNavigation-1 MuiGrid-item MuiGrid-grid-xs-12"
>
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root"
    data-testid="back-arrow"
    focusable="false"
    onClick={[Function]}
    viewBox="0 0 24 24"
  >
    <path
      d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
    />
  </svg>
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root"
    data-testid="forward-arrow"
    focusable="false"
    onClick={[Function]}
    viewBox="0 0 24 24"
  >
    <path
      d="M12 4l-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
    />
  </svg>
</div>
`;
